lua/nui/menu/init.lua

```diff
@@ -333,12 +333,22 @@ function Menu:mount()
   self._tree:render()
 
   -- focus first item
+  self:focus(1)
+end
+
+-- focus the item at given index, skipping items when needed
+---@return boolean? true when index was found and focused
+function Menu:focus(item_idx)
```

Comment:
I'm curious about the use-case for this. How would the caller know about the `item_idx`? 🤔

Reply:
For Noice, for example, I implement the native popupmenu. Through `ext_popupmenu`, I get `popupmenu_select` events, with an completion item index.

Other use cases for completion in general could be to preselect a given index when creating the menu. LSP servers can return a list of items, where it's not always the first one that has `preselect=true`

Reply:
Does the items has any unique id? I wonder if we should make `menu._tree` a public interface and document it. That would provide much more flexibility without having to add more methods to the core.

For example, targeting an item with arbritary filter would be possible (it's still possible, but exposing `menu._tree` as `menu.tree` and documenting it would make this more clear):

```
  local min_width = 3

  local items = {
    Menu.item("A", { id = "a" }),
    Menu.separator("*"),
    Menu.item("B", { id = "b" }),
  }

  local menu = Menu({
    relative = "win",
    position = "50%",
  }, {
    lines = items,
    min_width = min_width,
  })

  menu:mount()

  local function focus(id)
    local nodes = menu._tree:get_nodes()
    local target_node
    for _, node in ipairs(nodes) do
      if node.id == id then
        target_node = node
        break
      end
    end
    if target_node then
      local _, linenr = menu._tree:get_node(target_node:get_id())
      vim.api.nvim_win_set_cursor(menu.winid, { linenr, 0 })
    end
  end

  focus("b")
```

Reply:
Perfect, thanks!